src/util/bloom.rs

```diff
@@ -0,0 +1,1034 @@
+// Copyright (C) 2013-2020 Blockstack PBC, a public benefit corporation
+// Copyright (C) 2020-2021 Stacks Open Internet Foundation
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+// alternatives considered:
+//  * https://docs.rs/bloom/0.3.2/bloom/
+//      * Problems:
+//          * buggy -- has higher false-positive rate than expected
+//          * can't serialize/deserialize
+//          * can't induce a bloom filter from the counts
+//
+//  * https://github.com/crepererum/pdatastructs.rs
+//      * Problems:
+//          * No invertable bloom filter
+//          * Brings in lots of other code
+//
+//  * https://github.com/CodeChain-io/rust-iblt
+//      * Problems:
+//          * Can't control serialization/deserialization
+//          * Can't induce a bloom filter from the counts
+//          * Hilariously inefficient
```

Comment:
While useful to know what alternatives were considered, I'm not sure this should live on in the comment-- perhaps just in the PR body?

Reply:
ha, yup, removing